inference-engine/thirdparty/clDNN/src/gpu/loop_gpu.cpp

```diff
@@ -0,0 +1,453 @@
+        int actual_iteration = 0;
+        while (current_iteration < trip_count && execution_condition) {
+            // Copy & Set sliced input memory offset
+            for (size_t i = 0; i < instance.input_iteration_mem.size(); ++i) {
+                const auto& cropped_mem_info = input_iteration_mem.at(i);
+                memory_impl::ptr mem = cropped_mem_info.copy_and_get_sliced_mem(actual_iteration);
+                // set input mem
+                if (actual_iteration == 0) {
+                    body_network->set_input_data(cropped_mem_info.to_id, *mem);
+                } else {
+                    cropped_mem_info.to_prim->set_output_memory(*mem);
+                }
+            }
+
+            // Set backedged input
+            if (actual_iteration == 0) {
+                for (auto& edge_mem_bind : instance.backedge_mem) {
+                    const primitive_id& input_id = edge_mem_bind.to_primitive->id();
+                    body_network->set_input_data(input_id, *edge_mem_bind.initial_mem);
+                }
+            } else {
+                for (auto& edge_mem_bind : instance.backedge_mem) {
+                    edge_mem_bind.set_backedged_input(actual_iteration);
+                }
+            }
+
+            // Set sliced output memory offset
+            for (size_t i = 0; i < output_iteration_mem.size(); ++i) {
+                const auto& cropped_mem_info = output_iteration_mem.at(i);
+                const auto& from_mem = cropped_mem_info.sliced_mem.at(actual_iteration);
+                cropped_mem_info.from_prim->set_output_memory(*from_mem);
+            }
+            if (actual_iteration == 0) {
+                body_network->execute(events);
+            } else {
+                body_network->execute({});
+            }
+
+
+            // update index & execution condition for the next iteration
+            if (node.is_current_iteration_used()) {
+                current_iteration = read_int(*current_iteration_mem);
+                ++current_iteration;
+                write_int(*current_iteration_mem, current_iteration);
+            } else {
+                ++current_iteration;
+            }
+            if (node.is_execution_condition_used()) {
+                execution_condition = read_int(*execution_condition_mem);
+            }
+            ++actual_iteration;
+        }
+        body_network->reset_execution();
+
+        // Concatenate sliced output to the outer network
+        for (size_t i = 0; i < output_iteration_mem.size(); ++i) {
+            const auto& cropped_mem_info = output_iteration_mem.at(i);
+            cropped_mem_info.copy_to_concatenated_mem();
+        }
+
+        const primitive_id& num_iteration_id = node.get_num_iteration_id();
+        memory_impl& num_iteration_mem = outer_network.get_primitive(num_iteration_id)->output_memory();
+        write_int(num_iteration_mem, actual_iteration);
```

Comment:
Ummm.. I meant 
- Use this varialbe as "cur_iteration" as variable throughout the loop iteration, 
- Rename "num_iteration_mem" as "num_actual_iterations_mem" or "num_actual_seq_len_mem" 
- and write(int num_actual_seq_len_mem, cur_iteration) 

How do you think..